core/src/macosX64Main/kotlin/CentralManagerDelegate.kt

```diff
@@ -24,11 +22,8 @@ import kotlin.native.concurrent.freeze
 // https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerdelegate
 internal class CentralManagerDelegate : NSObject(), CBCentralManagerDelegateProtocol {
 
-    // todo: MutableSharedFlow
-    private val _onDisconnected = BroadcastChannel<NSUUID>(1)
-    internal val onDisconnected: Flow<NSUUID>
-        // todo: Drop `get()` when `_onDisconnected` is a MutableSharedFlow.
-        get() = _onDisconnected.openSubscription().consumeAsFlow()
+    private val _onDisconnected = MutableSharedFlow<NSUUID>()
+    internal val onDisconnected = _onDisconnected.asSharedFlow()
```

Comment:
Just curious, is a MutableSharedFlow not _already_ a shared flow?

Reply:
`MutableSharedFlow` is a mutable version of `SharedFlow`.

Using `asSharedFlow` exposes the underlying (`_onDisconnected`) **mutable** version of the flow, as a read-only `SharedFlow`.

Reply:
Gotcha. So privately you want it mutable and immutable internally.

Reply:
Ya, basically isolating that emissions on the `Flow` can only come from within this class (and that downstream consumers can never also push items onto the stream — can only read from the stream).

Basically, just good practice to help keep the overall system simpler and less error prone.